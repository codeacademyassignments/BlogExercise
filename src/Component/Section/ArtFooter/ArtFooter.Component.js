import React from 'react';
import '../../index.css';

const ArtFooter = (props) => {
  const heartUrl = props.liked ? 'heart-red.svg' : 'heart-black.svg';
  console.log(props);
  return (
    <>
      <img
        src={require('../../../Icons/clapping.svg')}
        alt="clapping"
        className="claps"
        onClick={(event) => { props.onClapClick(); }}
      />
      <span className="no-of-claps">{props.claps}</span>
      <img
        src={require(`../../../Icons/${heartUrl}`)}
        alt="heart-black"
        className="heart"
        onClick={() => {
          props.onHeartClick();
        }}
      />
    </>
  );
};

export default ArtFooter;
