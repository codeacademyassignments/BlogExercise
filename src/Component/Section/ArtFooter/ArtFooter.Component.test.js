import React from 'react';
import renderer from 'react-test-renderer';
import ArtFooter from './ArtFooter.Component';
import {shallow} from 'enzyme';


describe('snapshot test',()=>{
  it('snapshot should match',()=>{
    const tree = renderer.create(<ArtFooter liked={true} claps={10}/>);
    expect(tree).toMatchSnapshot();
  })
})

describe('mock function',()=>{
  const onClapClickMock = jest.fn();
  const onHeartClickMock = jest.fn();
  const wrapper = shallow(<ArtFooter liked={true} claps = {2} onClapClick={onClapClickMock} onHeartClick={onHeartClickMock}/>)
  it('should call onClapClickMock when clap clicked',()=>{
    wrapper.find('.claps').simulate('click');
    expect(onClapClickMock).toHaveBeenCalledWith();
    expect(onClapClickMock).toHaveBeenCalledTimes(1);
  })
  it('should call onHeartClickMock when clap clicked',()=>{
    wrapper.find('.heart').simulate('click');
    expect(onHeartClickMock).toHaveBeenCalledWith();
    expect(onHeartClickMock).toHaveBeenCalledTimes(1);
  })
})