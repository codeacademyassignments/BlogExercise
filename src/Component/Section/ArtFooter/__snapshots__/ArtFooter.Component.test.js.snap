// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test snapshot should match 1`] = `
Array [
  <img
    alt="clapping"
    className="claps"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "clapping.svg",
      }
    }
  />,
  <span
    className="no-of-claps"
  >
    10
  </span>,
  <img
    alt="heart-black"
    className="heart"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "heart-red.svg",
      }
    }
  />,
]
`;
